optaplanner-core/src/main/java/org/optaplanner/core/api/compatibility/lang/NonNull.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.api.compatibility.lang;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.annotation.Nonnull;
+import javax.annotation.meta.TypeQualifierNickname;
+
+/**
+ * Indicates that a field, method return value or method parameter is never null.
+ * This is useful for Kotlin users and code quality checks.
+ * <p>
+ * This is the opposite of {@link Nullable}.
+ */
+@Target({ElementType.METHOD, ElementType.PARAMETER, ElementType.FIELD})
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@Nonnull
+@TypeQualifierNickname
+public @interface NonNull {
```

Comment:
`@NotNull` please :slightly_smiling_face:

Reply:
The most common seems to be NonNull, but I 'll need to bring this up with the MW architect list, good point.

Reply:
1 point for `@NonNull`: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/lang/package-summary.html